fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -916,7 +917,7 @@ ACTOR Future<Void> getResolution(CommitBatchContext* self) {
 
 	// Fetch cipher keys if needed.
 	state Future<std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>>> getCipherKeys;
-	if (SERVER_KNOBS->ENABLE_TLOG_ENCRYPTION) {
+	if (isEncryptionOpSupported(EncryptOperationType::TLOG_ENCRYPTION, pProxyCommitData->db->get().client)) {
```

Comment:
I'm not familiar how the ClientDBInfo works. Is `isEncryptionEnabled` immutable without a restart? In that case how about we check `isEncryptionOpSupported` just once when CommitProxyServer starts? Probably same for BlobWorker.

Reply:
It should be but there is a case of provisional proxies which @sfc-gh-ahusain and I discussed where we might run into issues. Also if in the future we decide to use other fields of the clientDB object (other than just `isEncryptionEnabled`) to determine whether to turn off/on encryption (such as tenant features) then it may be useful not to just call this function on proxy start. In any case since this would be a major change to the pr I think it might be best to do it in a separate pr if we deem it necessary.

Reply:
Can you explain what the provisional proxies case is? 

One other thing is that, in commit proxy, for one commit batch, the `isEncryptionOpSupported` value should be the same. If the `isEncryptionEnabled` value can change dynamically, we need to change the code to just check `isEncryptionOpSupported` once for one commit batch.

Reply:
discussed offline

Reply:
> One other thing is that, in commit proxy, for one commit batch, the isEncryptionOpSupported value should be the same. If the isEncryptionEnabled value can change dynamically, we need to change the code to just check isEncryptionOpSupported once for one commit batch.

Good point, however, the value of the knob is determined at the bootstrap of the test (I believe); also, Yi you fixed the behavior where for encryption, restart tests carry forward the values. I think we should be good here.